nx-dev/data-access-ai/src/lib/data-access-ai.ts

```diff
@@ -0,0 +1,188 @@
+// based on:
+// https://github.com/supabase-community/nextjs-openai-doc-search/blob/main/pages/api/vector-search.ts
+
+import { createClient } from '@supabase/supabase-js';
+import { codeBlock, oneLine } from 'common-tags';
+import GPT3Tokenizer from 'gpt3-tokenizer';
+import {
+  Configuration,
+  OpenAIApi,
+  CreateModerationResponse,
+  CreateEmbeddingResponse,
+  ChatCompletionRequestMessageRoleEnum,
+} from 'openai-edge';
+import { OpenAIStream, StreamingTextResponse } from 'ai';
+
+const openAiKey = process.env['NX_OPENAI_KEY'];
+const supabaseUrl = process.env['NX_NEXT_PUBLIC_SUPABASE_URL'];
+const supabaseServiceKey = process.env['NX_SUPABASE_SERVICE_ROLE_KEY'];
+const config = new Configuration({
+  apiKey: openAiKey,
+});
+const openai = new OpenAIApi(config);
+
+export async function nxDevDataAccessAi(
+  query: any
```

Comment:
I think this is a `string` from base on the input's value.
```suggestion
  query: string
```